crates/ext-processor/src/service.rs

```diff
@@ -0,0 +1,1045 @@
+                }
+            }
+
+            return Ok(request.body(request_chunk)?);
+        }
+        Err(PrepareRequestError::MissingHeaders)
+    }
+
+    async fn prepare_response(
+        stream: &mut Streaming<ProcessingRequest>,
+    ) -> Result<bulwark_wasm_sdk::Response, PrepareResponseError> {
+        if let Some(header_msg) = Self::get_response_headers(stream).await {
+            let status = Self::get_header_value(&header_msg.headers, ":status")
+                .ok_or(PrepareResponseError::MissingStatus)?;
+
+            let mut response = http::Response::builder();
+            // TODO: read the response body
+            let response_chunk = bulwark_wasm_sdk::BodyChunk {
+                end_of_stream: header_msg.end_of_stream,
+                size: 0,
+                start: 0,
+                content: vec![],
+            };
+            response = response.status(status);
+            match &header_msg.headers {
+                Some(headers) => {
+                    for header in &headers.headers {
+                        // must not pass through Envoy pseudo headers here, http module treats them as invalid
+                        if !header.key.starts_with(':') {
+                            response = response.header(&header.key, &header.value);
+                        }
+                    }
+                }
+                None => {}
+            }
+            return Ok(response.body(response_chunk)?);
+        }
+        Err(PrepareResponseError::MissingHeaders)
+    }
+
+    fn instantiate_plugins(
+        plugins: &PluginList,
+        redis_info: Option<Arc<RedisInfo>>,
+        http_req: Arc<bulwark_wasm_sdk::Request>,
+        params: matchit::Params,
+    ) -> Result<Vec<Arc<Mutex<PluginInstance>>>, PluginGroupInstantiationError> {
+        let mut plugin_instances = Vec::with_capacity(plugins.len());
+        let mut shared_params = bulwark_wasm_sdk::Map::new();
+        for (key, value) in params.iter() {
+            let wrapped_value = bulwark_wasm_sdk::Value::String(value.to_string());
+            shared_params.insert(format!("param.{}", key), wrapped_value);
+        }
+        let shared_params = Arc::new(Mutex::new(shared_params));
+        for plugin in plugins {
+            let request_context = RequestContext::new(
+                plugin.clone(),
+                redis_info.clone(),
+                shared_params.clone(),
+                http_req.clone(),
+            )?;
+
+            plugin_instances.push(Arc::new(Mutex::new(PluginInstance::new(
+                plugin.clone(),
+                request_context,
+            )?)));
+        }
+        Ok(plugin_instances)
+    }
+
+    async fn execute_request_phase(
+        plugin_instances: Vec<Arc<Mutex<PluginInstance>>>,
+        timeout_duration: std::time::Duration,
+    ) -> DecisionComponents {
+        Self::execute_request_phase_one(plugin_instances.clone(), timeout_duration).await;
```

Comment:
Yeah, that choice is mainly a holdover from earlier in development, when the activity within a phase wasn't as well defined, but I expect a 3rd phase for handling the body. It could be clarified at this point.